framework/src/transfers/MultiAppUserObjectTransfer.C

```diff
@@ -60,14 +60,20 @@ MultiAppUserObjectTransfer::validParams()
       "Samples a variable's value in the Master domain at the point where the MultiApp is and "
       "copies that value into a post-processor in the MultiApp");
 
+  params.addParam<bool>("nearest_sub_app",
+                        false,
+                        "When True, a from_multiapp transfer will work by finding the nearest "
+                        "(using the `location`) sub-app and query that for the value to transfer");
+
   return params;
 }
 
 MultiAppUserObjectTransfer::MultiAppUserObjectTransfer(const InputParameters & parameters)
   : MultiAppConservativeTransfer(parameters),
     _user_object_name(getParam<UserObjectName>("user_object")),
     _all_master_nodes_contained_in_sub_app(getParam<bool>("all_master_nodes_contained_in_sub_app")),
-    _skip_bbox_check(getParam<bool>("skip_bounding_box_check"))
+    _skip_bbox_check(getParam<bool>("skip_bounding_box_check")),
+    _nearest_sub_app(getParam<bool>("nearest_sub_app"))
 {
   // This transfer does not work with DistributedMesh
   _fe_problem.mesh().errorIfDistributedMesh("MultiAppUserObjectTransfer");
```

Comment:
Are we able to remove this? I can not see any issue in this transfer.

Reply:
Just UO does not work with a distributed mesh?

Reply:
Unfortunately, no - this transfer _depends_ on replicated mesh.  It's because the UO's you're pulling from will be distributed in parallel in a different way from your mesh.

This can be fixed in the future - you would need to to use `pull_parallel_vector_data` to do queries for UO data.

But - that wasn't the point of this particular work.

Reply:
Just to cross-reference things... here is the Issue for making this work with DistributedMesh: #20099